### DISCUSSION TITLE
How to clear cached reference to old local ACTIVEPIVOT_LICENSE env variable?

### DISCUSSION THREAD

**Discussion by LeighTilley** [atoti/atoti#787] [Q&A] [answered]
Hi

I'm a seasoned Java developer who works with the Java ActivePivot on client site in day job.

I just got around to downloading the Atoti free CE to try some stuff at home. It appears to have picked up an old Windows env variable from ages ago (I had some test stuff setup a few years ago) for ACTIVEPIVOT_LICENSE.

I removed the variable and relaunched everything however it appears to have cached stuff. I know no Python at moment but did find the cache in AppData but not sure if I can delete it or not. 

How can I remove this caching so that it gets an appropriate new licence from somewhere? I seem to remember a LinkedIn post from few days ago about cloud licencing these days...

Thanks

Mr Leigh Tilley
https://tilleytech.com

**Comment by tibdex** [comment]
Removing the environment variable should have been enough. Can you try to run this script?

```python
import os

if "ACTIVEPIVOT_LICENSE" in os.environ:
  print("The environment variable has not been correctly removed.")
else:
  print("The environment variable has been removed correctly.")
```

**Comment by LeighTilley** [comment]
Hi

![image](https://github.com/atoti/atoti/assets/9533038/a6863bbd-a8d2-47d3-acd7-405de6ea1d39)

**Reply by tibdex** [reply]
Looks good.

Can you share the output of this script then?

```python
import sys

import atoti as tt
import pandas as pd

session = tt.Session(logging=tt.LoggingConfig(destination=sys.stdout))

dataframe = pd.DataFrame(
    {
        "City": ["Paris", "New York", "London", "Paris", "New York", "London"],
        "Color": ["blue", "blue", "blue", "red", "red", "red"],
        "Price": [10, 20, 30, 40, 50, 60],
    }
)

table = session.read_pandas(dataframe, keys=["City", "Color"], table_name="Prices")
cube = session.create_cube(table)

```

**Comment by LeighTilley** [comment]
Hi

So if I run Jupyter Lab it opens with the errors, with final part below. If I close it, without saving, and re-open it then it is already at the error point so it seems to create some sort of checkpoint. 

The environment variable isn't even there so as you can see it is using some sort of cache.

![image](https://github.com/atoti/atoti/assets/9533038/bbbbc8f4-5ad4-4cff-a198-4521b1d578d8)

**Reply by tibdex** [reply]
You ran the script from https://github.com/atoti/atoti/discussions/787#discussioncomment-6107975 in a new terminal. Just to be 100% sure, can you also run it in your Jupyter notebook? Maybe the environment variable is still picked up by Jupyter...

Actually, you can run this script that covers everything:

 ```python
import os

for name in ["ACTIVEPIVOT_LICENSE", "ATOTI_LICENSE"]:
  if name in os.environ:
    print(f"The {name} environment variable is defined.")
  else:
    print(f"No {name} environment variable defined.")

import sys

import atoti as tt
import pandas as pd

session = tt.Session(logging=tt.LoggingConfig(destination=sys.stdout))

dataframe = pd.DataFrame(
    {
        "City": ["Paris", "New York", "London", "Paris", "New York", "London"],
        "Color": ["blue", "blue", "blue", "red", "red", "red"],
        "Price": [10, 20, 30, 40, 50, 60],
    }
)

table = session.read_pandas(dataframe, keys=["City", "Color"], table_name="Prices")
cube = session.create_cube(table)
```

**Comment by LeighTilley** [comment]
Hi

OK I found a terminal in Jupyter (this is the first time ive used any of it so I do not yet know my way around :)).

![image](https://github.com/atoti/atoti/assets/9533038/e9ecfe8a-d4ca-42d7-b06f-9383e7ade4f0)

**Reply by rcolle** [reply]
Hi,

I don't mean to be rude, but can you please consider going through a basic Python and Jupyter training?
From your previous posts, it looks like you have never gone through the basics of the language and tooling.

If the problem still exists after that, let's continue this discussion.

Thanks!
Romain

**Reply by LeighTilley** [reply, marked as answer]
Hi

Thanks although as the title of the post states, and the original picture, it was picking up an old env variable related to an original  Java setup.

I got a chance to check it again this morning, running:

![image](https://github.com/atoti/atoti/assets/9533038/a2dab5e4-1410-45fc-9948-de3ebbd33878)

And I also removed the local old lic file too. Now it's moved along to only use Atoti stuff (to the below) so as the error message says I need to update so will do so :):

![image](https://github.com/atoti/atoti/assets/9533038/b56f5991-1b4a-4af1-8ad3-2c4010a905f7)

Thanks

Leigh

**Comment by LeighTilley** [comment]
Hey

OK yes so will get this upgraded:

![image](https://github.com/atoti/atoti/assets/9533038/baa08c0f-e157-473f-a3ab-6e8a06aba369)

**Comment by LeighTilley** [comment]
Hi

OK updated to 0.8.0 and had to manually remove jupyterlab 0.6..6 (some warnings / conflicts even with no cache etc)

Now creates:

![image](https://github.com/atoti/atoti/assets/9533038/02042d1b-f13f-4fca-a55e-1f184fa9d29d)


thx
Mr Leigh Tilley
https://tilleytech.com/

**Comment by LeighTilley** [comment]
Hi

Yes all good now. Tested outside of Jupyter via Excel too (as this all actually for something much more custom!)

![image](https://github.com/atoti/atoti/assets/9533038/b5176bd2-b249-4596-9bdd-a3da17852f88)
